Lesson 6

dimakonoplev opened the pull request:
Александр, прошу оценить дом. работу

dimakonoplev committed:
создание новой иерархии классов в связи с повторяющимся кодом

dimakonoplev committed:
продолжаем создавать новую еирархию классов

dimakonoplev committed:
no commit

dimakonoplev committed:
No commit

dimakonoplev committed:
delete comment

dimakonoplev committed:
Удалил ненужную ф-ию

dimakonoplev committed:
удалил комментарий

dimakonoplev committed:
удалил коммент

dimakonoplev committed:
закоментил строку кода

dimakonoplev committed:
еще Map

dimakonoplev committed:
сортировка карты частот слов

dimakonoplev committed:
текст

dimakonoplev committed:
Сортировка частот слов по убыванию ключа

dimakonoplev committed:
Топ 10 слов
количество уникальных слов

time2die commented on src/lessons/homework/Task6.java:
```diff
@@ -0,0 +1,81 @@
+        stringList.addAll(ReadInput.readText());
+
+        //- посчитать частотность повторения всех слов и вывести на экран топ 10
+        String strArray = stringList.toString();// List -> String
```
максимальный размер String int, мы можем потерять данные

time2die commented on src/lessons/homework/Task6.java:
```diff
@@ -0,0 +1,81 @@
+        String[] stringArray =  strClear.trim().split("\\s+");// разбить строку на слова
+
+        //-------------
+        ArrayList<String> arrayList = new ArrayList<>(Arrays.asList(stringArray));// String -> List
```
данные аналогичны stringList

dimakonoplev committed:
Поменял тип String на ArrayList, теперь нет вероятности потерять данные

time2die commented on src/lessons/crossline/CrossLineMain.java:
```diff
@@ -0,0 +1,102 @@
+            makeStep(p1, gameArrayCharPole);
+            if (!gameArrayCharPole.hasCleanCell()) {
+                System.out.println("nobody win");
+                System.exit(0);
```
код же повторяется

dimakonoplev replied:
Да, точно. Можно оптимизировать.

dimakonoplev committed:
добавил коментарии

dimakonoplev committed:
Убрал повторяющийся код в функцию.

dimakonoplev committed:
Удалил лишнее

dimakonoplev committed:
еще раз

dimakonoplev committed:
Хм

dimakonoplev committed:
Убрал повторяющийся код

dimakonoplev committed:
добавил функци:
 - Разбивка на символы
 - Создание частотной карты слов
 - Создание частотной карты символов

dimakonoplev commented:
Программа выполняется долго, секунд 30

dimakonoplev committed:
Немного усовершенствовал

kemsta commented on src/lessons/homework/Task6.java:
```diff
@@ -0,0 +1,235 @@
+        // значение Карты - частота встречаемости каждого ключа данной Карты (значение setOne) в listOne
+        for(Iterator<String> iter = setOne.iterator(); iter.hasNext();){// проходимся итератором по множеству setOne
+            String val = iter.next();// сохраним очередное значение множества setOne...
+            int freq = Collections.frequency(arrayList, val);// сохраним его частоту
```
Collections.frequency executed 21188 time, with full time execution 12950ms and average time execution 611215ns
Collections.frequency итерирует каждый раз всю коллекцию, т.е. по сути вы выполняете 21188 итерации внутри каждой из которых выполняется 21188 итераций = 448931344  проходов по мапе. А еще Collections.frequency считает null как совпадающее с любым значением и увеличивает счетчик.

dimakonoplev replied:
Спасибо за подсказку. Попробую исправить

dimakonoplev committed:
изменил структура кода, для удобаства добавил новый класс work6_1 и перенес туда все фуннкции

dimakonoplev committed:
--

dimakonoplev committed:
--

dimakonoplev committed:
добавил еще один класс Work6_2 для усовершенствования кода из класса Work6_1

dimakonoplev committed:
--

dimakonoplev committed:
добавил более быстрые функции - код работает намного быстрее
